### ISSUE TITLE
Login only after multiple attempts | Using Subdomains and PG

### ISSUE THREAD

**Issue by shreyaq3** [plataformatec/devise#3904]
Hi,

I am rails newbie and I'm facing a weird issue. I'm using rails 3.2 and devise 3.0
I'm using subdomains and PG's multiple schemas to segregate one account from another .

Users on my system are able to login only after multiple attempts. There is nothing in the log and I'm absolutely clueless. I hope someone can point me in the right direction.

My code is as follows
Application Controller - 
 
```
  before_filter :setup_schema

  def setup_schema
    if request.subdomain.blank? || request.subdomain == 'www'
      redirect_to "http://www.#{ENV['DOMAIN']}.com"
    else
      @tenant = Account.find_by_name(request.subdomain)
      if @tenant.nil?
        redirect_to "http://www.#{ENV['DOMAIN']}.com"
      else
        ActiveRecord::Base.connection.schema_search_path = "#{@tenant.schema},hstore"
        Delayed::Job.table_name = 'public.delayed_jobs'
      end
    end
  end
```

Session Store -
```
XXX::Application.config.session_store :cookie_store, key: '_XXX_session', domain: :all
```

routes
```
  devise_for :users, controllers: {registrations: "registrations"}, path_names: { sign_in: "login", sign_out: "logout"}
  as :user do
    get 'users/edit' => 'devise/registrations#edit', as: 'edit_user_registration'
    get 'users/edit_password' => 'devise/registrations#edit_password', as: 'edit_pass_user_registration'
    post "user_create" => "users#create",  as: 'user_create'
  end

  resources :dashboards, only: [:index]

  devise_scope :user do
    constraints(Subdomain) do
      authenticated :user do
        root to: 'dashboards#index'
      end
      unauthenticated :user do
        root to: 'sessions#new'
      end
      root to: 'dashboards#index'
    end
  end
````
Another issue that I'm facing is - 
Lets say I'm logged in 1 subdomain and I'm at root path, ie, subdomain1.XXX.com and then I login with another user in another subdomain in another browser, now when I come to my first subdomain, I get logged out.
However, if I was a some other url eg, subdomain1.XXX.com/projects and then I login with another user in another subdomain in another browser, now when I come to my first subdomain, I DONT get logged out.

Can some point what is it that I'm missing?
I'm missing some key concept or this has to with devise?

**Comment by josevalim** [comment]
Please use the mailing list or StackOverflow for questions/help, where a wider community will be able to help you. We reserve the issues tracker for issues only.
